### PR TITLE
Melodic compatibility: rewrite getifip to make use of iproute2

### PR DESCRIPTION
**Pull request by Rayman**
ifconfig is deprecated in favor of iproute2. So I rewrote the script to parse the json output of the `ip route addr show <INTERFACE>` command.

This should also work for lower versions of ubuntu, but I did not test those.

### CODE REVIEW

File: scripts/getifip [hunk 1]
```diff
@@ -1,30 +1,23 @@
...
+from __future__ import print_function
 
-echo `LANG=C ifconfig $1 | grep -o 'inet addr:[^ ]*' | cut -d: -f2`
+import json
+import shlex
+import subprocess
+from argparse import ArgumentParser
+
+parser = ArgumentParser(description='Process some integers.')
+parser.add_argument('interface', help='an integer for the accumulator')
```
Comment (mikepurvis) [inline scripts/getifip]: Please update these strings.

File: scripts/getifip [hunk 2]
```diff
@@ -1,30 +1,23 @@
...
-# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-# DISCLAIMED. IN NO EVENT SHALL CLEARPATH ROBOTICS, INC. BE LIABLE FOR ANY
-# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
-# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
-# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
-# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-# 
-# Please send comments, questions, or patches to code@clearpathrobotics.com
```
Comment (mikepurvis) [inline scripts/getifip:28]: Please leave the BSD notice here, but feel free to change the copyright to yourself or add your own.

File: scripts/getifip [hunk 3]
```diff
@@ -1,30 +1,23 @@
...
+import subprocess
+from argparse import ArgumentParser
+
+parser = ArgumentParser(description='Process some integers.')
+parser.add_argument('interface', help='an integer for the accumulator')
+
+args = parser.parse_args()
+
+cmd = shlex.split("ip -json -4 addr show") + [args.interface]
+output = subprocess.check_output(cmd)
```
Comment (mikepurvis) [inline scripts/getifip:43]: I guess it's style, but I would avoid shlex here. Just list the five separate chunks in a list or tuple and avoid the concatenation as well.

**Review by mikepurvis — CHANGES_REQUESTED** [review]
Great, thanks for this contribution!

**Comment by Rayman** [comment]
Thanks you for the review. I've addressed all comments.

File: scripts/getifip [hunk 4]
```diff
@@ -1,17 +1,45 @@
 #!/usr/bin/env python
+#
+# Author: Ramon Wijnands <ramon@ruvu.nl>
+#         Copyright (c) 2090, RUVU, b.v.
```
Comment (mikepurvis) [inline scripts/getifip]: 2019?

Comment (gavanderhoorn) [reply scripts/getifip]: No this is from the future ;)

Comment (Rayman) [reply scripts/getifip]: Fixed this

**Comment by Rayman** [comment]
This patch only seems to work on melodic because `iproute2` should support the -json option. Can you make a melodic-devel branch then I'll retarget my PR.

**Comment by mikepurvis** [comment]
I'm not over the moon about needing a distro branch in this repo, just for this change. Would it be possible to support both? Like, perhaps by testing for `ip addr` supporting the `-json` flag, and if not, falling back on the other `ifconfig` invocation?

For the purposes of the fallback, I'd be fine if you just wanted to shell out to a `bash -c` rather than re-implementing the pipes using a bunch of tortured subprocess stuff.

**Comment by Rayman** [comment]
I've done another try with this PR: https://github.com/clearpathrobotics/robot_upstart/pull/88. This keeps a single branch for all distros.

**Pull request closed without merging**